extensions/amp-analytics/0.1/test/test-video-analytics.js

```diff
@@ -0,0 +1,602 @@
+      win.document.body.appendChild(analyticsElement);
+
+      target = win.document.createElement('div');
+      target.classList.add('target');
+      win.document.body.appendChild(target);
+
+      myVideo = win.document.createElement('amp-video');
+      myVideo.setAttribute('id', 'myVideo');
+      target.appendChild(myVideo);
+
+      myVideo2 = win.document.createElement('amp-video');
+      myVideo2.setAttribute('id', 'myVideo-2');
+      target.appendChild(myVideo2);
+    });
+
+    describe('VideoEventTracker', () => {
+      let tracker;
+      const selectors = ['#myVideo', '#myVideo-2'];
+
+      const defaultVideoConfig = {
+        'on': 'video-play',
+        'selector': selectors,
+        'videoSpec': {
+          'end-session-when-invisible': false,
+        },
+      };
+
+      beforeEach(() => {
+        tracker = root.getTracker(AnalyticsEventType.VIDEO, VideoEventTracker);
+      });
+
+      it('should initalize, add listeners and dispose', () => {
+        const fn1 = env.sandbox.stub();
+        expect(tracker.root).to.equal(root);
+        expect(tracker.sessionObservable_.getHandlerCount()).to.equal(0);
+
+        tracker.add(
+          undefined,
+          AnalyticsEventType.VIDEO,
+          defaultVideoConfig,
+          fn1
+        );
+
+        expect(tracker.sessionObservable_.getHandlerCount()).to.equal(1);
+
+        tracker.dispose();
+
+        expect(tracker.sessionObservable_).to.equal(null);
+      });
+
+      it('should require selector', () => {
+        allowConsoleError(() => {
+          expect(() => {
+            tracker.add(analyticsElement, AnalyticsEventType.VIDEO, {
+              selector: '',
+            });
+          }).to.throw(/Missing required selector/);
+        });
+      });
+
+      it('should error on duplicate selectors', () => {
+        const config = {
+          selector: ['#myVideo', '#myVideo'],
+        };
+
+        expect(() => {
+          tracker.add(
+            analyticsElement,
+            AnalyticsEventType.VIDEO,
+            config,
+            env.sandbox.stub()
+          );
+        }).to.throw(
+          /Cannot have duplicate selectors in selectors list: #myVideo,#myVideo/
+        );
+      });
+
+      it('fires on one video selector', async () => {
+        const fn1 = env.sandbox.stub();
+        const getElementSpy = env.sandbox.spy(root, 'getElement');
+
+        tracker.add(
+          undefined,
+          AnalyticsEventType.VIDEO,
+          {
+            'on': 'video-play',
+            'request': 'event',
+            'selector': '#myVideo',
+          },
+          fn1
+        );
+
+        const data = {
+          'autoplay': false,
+          'currentTime': 0.0045,
+          'duration': 15,
+          'height': 399,
+          'id': 'myVideo',
+          'muted': false,
+          'playedTotal': 0.0045,
+          'playedRangesJson': '[[0,0.0045]]',
+          'state': 'playing_manual',
+          'width': 700,
+        };
```

Comment:
Are these `data` objects we're creating all the same or very similar? If so, can we create a helper method to generate them for us?

Reply:
I moved the definition of the object to beforeEach so that all tests can use the same one. The data object for percentage played is a little different so I defined a separate one for it.